hal/shared/cellular_enums_hal.h

```diff
@@ -67,7 +67,7 @@ typedef struct {
     char meid[18+1];    //!< Mobile Equipment IDentifier
     char manu[16];      //!< Manufacturer (u-blox)
     char model[16];     //!< Model Name (LISA-U200, LISA-C200 or SARA-G350)
-    char ver[16];       //!< Software Version
+    char ver[16+1];       //!< Software Version
```

Comment:
This also does not resolve the issue for a case where we are running an old application (without this change yet) and system part with this fix. This is why normally for ABI stability we have `size` and/or `version` fields in structs that go through dynalib interface.

We are also not really fixing the root cause of the problem as you've already pointed out: `_cbString` has no idea of the size of the buffer it was passed. This should be trivially fixable by passing a pointer to a stack-allocated struct with `char*` and size, instead of just `char*`.

```c++
// Somewhere
struct CStringHelper {
    CStringHelper(char* s, size_t sz)
        : str{s},
          size{sz} {
    }
    CStringHelper() = delete;

    char* str;
    size_t size;
};

// ...
CStringHelper str(_dev.ver, sizeof(_dev.ver));
sendFormated("AT+CGMR\r\n");
if (RESP_OK != waitFinalResp(_cbString, &str)) {
    // ...
// ...

int MDMParser::_cbString(int type, const char* buf, int len, CStringHelper* str)
{
    if (str && (type == TYPE_UNKNOWN)) {
        char format[32] = {};
        snprintf(format, sizeof(format), "\r\n%%%us\r\n", str->size);
        if (sscanf(buf, format, str->str) == 1)
            /*nothing*/;
    }
    return WAIT;
}
```

Reply:
I like Andrey's suggestion here, and it looks good to me.  @keeramis would you be willing to test this approach?  We would update the PR to patch all _cbString callback calls as well.

Reply:
Also made fixes in other places where strings sizes are probably not checked. Please check..

And tested that values are getting stored into the `_dev.variable` correctly as well.